### PR TITLE
core: drivers: fix random number reading errors in hisi_trng

### PR DESCRIPTION
**Pull request by loubaihui**
Fixes error use of in IO_READ32_POLL_TIMEOUT() macro in hisi_trng

### CODE REVIEW

**Review by jenswi-linaro — COMMENTED** [review]
Please be more specific in the commit message. 
A "Fixes:" tag pointing to the commit introducing the problem would be nice too.

File: core/drivers/hisi_trng.c [hunk 1]
```diff
@@ -71,7 +72,7 @@ static TEE_Result trng_init(void)
 {
 	TEE_Result ret = TEE_ERROR_GENERIC;
 
-	IMSG("TRNG driver init start.");
+	DMSG("TRNG driver init start.");
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/hisi_trng.c:75]: by the way, could remove the `.` char, here and for all `xMSG()` occurrences in this source file.

**Review by etienne-lms — COMMENTED** [review]
Please add `Fixes: fb5592f9cfeb ("core: drivers: add HiSilicon TRNG implementation")` tag in the commit message.

Could you also append a commit to add a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file? It would have catch the issues earlier.

File: core/drivers/hisi_trng.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (loubaihui) [reply core/drivers/hisi_trng.c:75]: > Unrelated change?
Use `DMSG` may be better ?

[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/hisi_trng.c:75]: This is not a bug, as the commit description mentions. 
It's all fine you make that change but prefer in a separated commit. I think this is what @jenswi-linaro had in mind.

[the remark this answers is not in the document]
Comment (loubaihui) [reply core/drivers/hisi_trng.c:75]: > This is not a bug, as the commit description mentions. It's all fine you make that change but prefer in a separated commit. I think this is what @jenswi-linaro had in mind.
Ok, thank you!

**Comment by loubaihui** [comment]
> Please add `Fixes: fb5592f9cfeb ("core: drivers: add HiSilicon TRNG implementation")` tag in the commit message.
> 
> Could you also append a commit to add a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file? It would have **~catch~ caught** the issues earlier.
> 
> **(edited)**

**Comment by loubaihui** [comment]
> Please add `Fixes: fb5592f9cfeb ("core: drivers: add HiSilicon TRNG implementation")` tag in the commit message.
> 
> Could you also append a commit to add a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file? It would have **~catch~ caught** the issues earlier.
> 
> **(edited)**

Does this need to submit in a new pull request?

**Comment by etienne-lms** [comment]
> Does this need to submit in a new pull request?

No. You only need to push fixup commits on the same branch you used to created this P-R. Once reviewers are happy, you'll be asked to squash the fixup commits, maybe update commit's messages if needed for a final review before you can applied posted review tags.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for the series with a fix in commit message for commit "Fixes error use of in IO_READ32_POLL_TIMEOUT() macro in hisi_trng":
```patch
-Fixes error use of in IO_READ32_POLL_TIMEOUT() macro in hisi_trng
+Fixes arguments passed to IO_READ32_POLL_TIMEOUT() macro
+and missing local variable definition in hisi_trng driver.

**Review by etienne-lms — CHANGES_REQUESTED** [review]
Could you add a 3rd commit that adds a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file?

**Comment by loubaihui** [comment]
> Could you add a 3rd commit that adds a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file?

Ok

**Comment by jforissier** [comment]
> Could you add a 3rd commit that adds a `_make PLATFORM=d06` command in _.github/workflows/ci.yml_ file?

Good suggestion it seems :wink: 

```
_make PLATFORM=d06
In file included from core/arch/arm/include/arm.h:11,
                 from core/arch/arm/include/kernel/delay_arch.h:33,
                 from core/include/kernel/delay.h:33,
                 from core/include/io.h:9,
                 from core/drivers/hisi_trng.c:5:
core/drivers/hisi_trng.c: In function 'hw_get_random_bytes':
lib/libutils/ext/include/util.h:28:13: error: comparison of integer expressions of different signedness: 'int' and 'size_t' {aka 'long unsigned int'} [-Werror=sign-compare]
   28 |          _a < _b ? _a : _b; }))
      |             ^
core/drivers/hisi_trng.c:63:24: note: in expansion of macro 'MIN'
   63 |                 size = MIN(HTRNG_BYTES, len - current_len);
      |                        ^~~
```

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commits
"core: drivers: modify debug and error messages in hisi_trng" and
"ci.yml: adds a PLATFORM=d06 command".

**Comment by etienne-lms** [comment]
And do not forget to rebase your changes on top of latest master branch. It should resolve the IBART CI test failure.

**Comment by loubaihui** [comment]
@jforissier

**Comment by etienne-lms** [comment]
CI make (multi-platform) test failed:
```
_make PLATFORM=d06
core/drivers/hisi_trng.c: In function 'hw_get_random_bytes':
core/drivers/hisi_trng.c:64:17: error: implicit declaration of function 'memcpy' [-Werror=implicit-function-declaration]
   64 |                 memcpy((uint8_t *)buf + current_len, &val, size);
      |                 ^~~~~~
core/drivers/hisi_trng.c:11:1: note: include '<string.h>' or provide a declaration of 'memcpy'
   10 | #include <rng_support.h>
  +++ |+#include <string.h>
   11 |

**Pull request merged**